### PR TITLE
Fixes #19037: Refactor the system techniques by component

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19037

### CODE REVIEW

File: techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    pass3.!result_error.!result_repaired::
+      "result_na"  expression => "any";
+
+  methods:
+      "restart_jetty_password" usebundle => disable_reporting;
+      "restart_jetty_password" usebundle => _method_reporting_context("Reload rudder services", "None");
+    rudder_system_restart_jetty::
+      "restart_jetty_password" usebundle => service_restart("${jetty_service_name}");
+    rudder_system_restart_apache::
+      "restart_jetty_password" usebundle => service_reload("${apache_service}");
```
Comment (ncharles) [inline techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf:40]:
```suggestion
      "restart_apache_password" usebundle => service_reload("${apache_service}");
```

File: techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  vars:
+      "jetty_service_name"  string => "rudder-jetty";
+      "apache_service_name" string => "apache2";
+      "relayd_service_name" string => "rudder-relayd";
+    redhat::
+      "apache_service_name" string => "httpd";
+
+    any::
+      "jetty_prefix"         string => canonify("service_restart_${jetty_service_name}");
+      "apache_prefix"         string => canonify("service_restart_${apache_service_name}");
```
Comment (ncharles) [inline techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf:11]:
```suggestion
      "apache_prefix"         string => canonify("service_reload_${apache_service_name}");
```

File: techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      "restart_jetty_password" usebundle => service_reload("${apache_service}");
+    rudder_system_restart_relayd::
+      "restart_jetty_password" usebundle => service_reload("${relayd_service}");
+    pass3::
+      "restart_jetty_password" usebundle => enable_reporting;
+
+      # Reporting
+      "report_error"   usebundle  => rudder_common_report("${technique_name}", "result_error", "${server_roles_common.directiveId}", "${component_name}", "None", "${report_string}"),
+                       ifvarclass => "result_error";
+      "report_repaired" usebundle => rudder_common_report("${technique_name}", "result_error", "${server_roles_common.directiveId}", "${component_name}", "None", "${report_string}"),
```
Comment (ncharles) [inline techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf:49]:
```suggestion
      "report_repaired" usebundle => rudder_common_report("${technique_name}", "result_repaired", "${server_roles_common.directiveId}", "${component_name}", "None", "${report_string}"),
```

File: techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      "restart_jetty_password" usebundle => service_restart("${jetty_service_name}");
+    rudder_system_restart_apache::
+      "restart_jetty_password" usebundle => service_reload("${apache_service}");
+    rudder_system_restart_relayd::
+      "restart_jetty_password" usebundle => service_reload("${relayd_service}");
+    pass3::
+      "restart_jetty_password" usebundle => enable_reporting;
+
+      # Reporting
+      "report_error"   usebundle  => rudder_common_report("${technique_name}", "result_error", "${server_roles_common.directiveId}", "${component_name}", "None", "${report_string}"),
```
Comment (ncharles) [inline techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf:47]: where is the report_string defined ?

File: techniques/systemSettings/systemRelay/1.0/relayd/relayd.cf [hunk 5]
```diff
@@ -0,0 +1,25 @@
+bundle agent rudder_system_relayd_configuration {
+  vars:
+      "config_dir"         string => "${g.rudder_base}/etc/relayd";
+      "config_file"        string => "${config_dir}/main.conf2";
```
Comment (ncharles) [inline techniques/systemSettings/systemRelay/1.0/relayd/relayd.cf:4]:
```suggestion
      "config_file"        string => "${config_dir}/main.conf";
```

**Review by ncharles — CHANGES_REQUESTED** [review]
this is really nicer than what we have
some minor changes request but this is a great step in the right direction

File: techniques/systemSettings/systemRelay/1.0/apache/main.cf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+      # Check that apache is running and enabled
+      "any" usebundle => _method_reporting_context("Apache service started", "None");
+      "any" usebundle => service_started("${apache_service}");
+
+      "any" usebundle => _method_reporting_context("Apache service enabled", "None");
+      "any" usebundle => service_enabled("${apache_service}");
+
+      # Configure relayd
+      "any" usebundle => rudder_system_relayd_configuration;
```
Comment (amousset) [inline techniques/systemSettings/systemRelay/1.0/apache/main.cf:29]: relayd should not be part of an apache bundle IMHO

Comment (Fdall) [reply techniques/systemSettings/systemRelay/1.0/apache/main.cf:29]: Sure, do you think I should make two different techniques (one apache and one relayd) or keep one "relay" technique with an apache and a relayd bundle?

File: techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    pass3.!result_error::
+      "result_repaired" expression => "${prefixes}_repaired";
+    pass3.!result_error.!result_repaired::
+      "result_na"  expression => "any";
+
+  methods:
+      "restart_jetty_password" usebundle => disable_reporting;
+      "restart_jetty_password" usebundle => _method_reporting_context("Reload rudder services", "None");
+    rudder_system_restart_jetty::
+      "restart_jetty_password" usebundle => service_restart("${jetty_service_name}");
```
Comment (amousset) [inline techniques/systemSettings/systemRelay/1.0/common/reload_rudder_services.cf:38]: no need to mention _password here, we do not need to leak this information across bundles, knowing we need to restart it is enough

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
Migrating the changes to https://github.com/Normation/rudder-techniques/pull/1655

**Pull request closed without merging**